src/utils/types.ts

```diff
@@ -0,0 +1 @@
+export type Url = string;
```

Comment:
do we need this type if it's just a string? just wondering because this will add an additional step when debugging to see what the `Url` type is

Reply:
maybe this can work together with a type-guard `isUrl` which receives a string, checks if the string is an actual url and then have the return type as `Url`

Reply:
I removed this for now as it's a distraction. Let's just get this big boy out of the door!